lib/datadog/core/configuration/base.rb

```diff
@@ -91,6 +91,7 @@ def dig(*options)
 
           def reset!
             reset_options!
+            Datadog::AppSec.settings.send(:reset!)
```

Comment:
Thanks for doing this change. Just in case, let me ask -- are we sure we can _always_ call this method on `Datadog::Appsec`?

E.g. I seem to remember a few cases where the constants were not loaded in some cases and we need to add a few `defined?(...)` checks.

Reply:
The only case I can think of is when people follow the [CI documentation ](https://docs.datadoghq.com/continuous_integration/tests/ruby/?tab=azurepipelines#instrumenting-your-tests) since they only require `datadog/ci`. 

I'm not sure how many times people would call `reset!` on those cases, but I will add the `defined?` just in case

Reply:
Done [599b534](https://github.com/DataDog/dd-trace-rb/pull/2910/commits/599b5342550a7293ad8f868a82190b3780b67af2)